freshmaker/handlers/botas/botas_shipped_advisory.py

```diff
@@ -66,11 +71,85 @@ def handle(self, event):
+                            " 'bundle_path_digests' set")
+                continue
+            bundle_digests.append(bundle['bundle_path_digest'])
+        bundle_images = self._pyxis.get_images_by_digests(bundle_digests)
+
+        # now we have nvrs of bundles to rebuild
+        bundles_nvrs = self._filter_bundle_images_by_builds(  # noqa: F841
+            bundle_images, rebuilt_images_digests)
 
         # Skip that event because we can't proceed with processing it.
         # TODO
-        # Next step would be queries Pyxis and getting the digests of these nvrs
+        # Now when we have bundle images' nvrs we can procceed with rebuilding it
+        msg = "Skipping due to being blocked on further implementation for now."
         db_event.transition(EventState.SKIPPED, msg)
         db.session.commit()
         return []
+
+    def _filter_bundle_images_by_builds(self, bundle_images, digests):
+        """
+        Filter out bundle images that don't have OSBS-pinning
+
+        :param dict bundle_images: Mapping of bundle digest to list of bundle's
+        ContainerImages
+        :param list digests: Digests of the rebuilt images that should be in
+        'related_images' and be 'pinned'
+        :return: list of nvrs of bundle images that have OSBS-pinning
+        """
+        ret_bundle_images_nvrs = set()
+        with koji_service(conf.koji_profile, log, dry_run=conf.dry_run,
+                          login=False) as session:
+            for bundle in bundle_images:
+                nvr = bundle['brew']['build']
+                build = session.get_build(nvr)
+                related_images = (
+                    build.get("build", {})
+                         .get("extra", {})
+                         .get("image", {})
+                         .get("operator_manifests", {})
+                         .get("related_images", {})
+                )
+                # skip bundle if it's not created by OSBS
+                if not related_images.get("created_by_osbs", False):
+                    continue
+
+                for pullspec in related_images.get('pullspecs', []):
+                    related_image_digest = self.parse_digest(pullspec['new'])
+                    if related_image_digest not in digests:
+                        continue
+
+                    # this is our old image, check if OSBS-pinning is
+                    # used for this image
+                    if pullspec.get("pinned", False):
+                        # add bundle nvr to a list for later handling
+                        ret_bundle_images_nvrs.add(nvr)
+                        break
+        return ret_bundle_images_nvrs
+
+    @staticmethod
+    def parse_digest(pullspec):
+        """
+        Helper method to parse digest from pullspecs.
+        """
+        digest = pullspec
+        if '@' in pullspec:
+            digest = pullspec.split('@', 1)[1]
+
+        return digest
```

Comment:
Are you sure you want to return `digest` even if there is no `@` in the pullspec? I'd suggest returning `None` and have the calling code handle that case.